blockchain/chain.go

```diff
@@ -1389,6 +1403,21 @@ func (b *BlockChain) BlockHeightByHash(hash *chainhash.Hash) (int32, error) {
 	return node.height, nil
 }
 
+// NodeHeightByHash returns the height of the block with the given hash in the
+// main chain. Does not check if the node is on the main chain, so returns height
+// of the node which is created by header validation
+//
+// This function is safe for concurrent access.
+func (b *BlockChain) NodeHeightByHash(hash *chainhash.Hash) (int32, error) {
```

Comment:
Why do we need  this function? This is only called in test code and you can do `node := LookupNode` anyways.

Reply:
This right now is being used while testing only, but it is also used to get height of the headers, when we are executing sliding block window logic.